source/AI.cpp

```diff
@@ -1468,6 +1469,8 @@ void AI::MoveIndependent(Ship &ship, Command &command) const
 		{
 			for(const System *link : links)
 			{
+				if(!unconstrained && !gov->AllowJumpingTo(*link))
```

Comment:
Good question, well if its empty its not expensive.
Otherwise it'll only check this if it has no system assigned, so I think it only happens once, except if the ship can't find a destination.
I could always add a bool to designate the ship as "unableToLeave" in that case.